.github/workflows/test.yml

```diff
@@ -0,0 +1,59 @@
+name: Run Tests
+on:
+  push:
+    branches: [ '*' ]
+  pull_request:
+    branches: [ main ]
+  schedule:
+    # Run automatically at 8AM PST Monday-Friday
+    - cron: '0 15 * * 1-5'
+  workflow_dispatch:
+
+jobs:
+  tests:
+    name: Build & Test
+    runs-on: ubuntu-latest
+    timeout-minutes: 20
+    strategy:
+      matrix:
+        go: [ '1.14', '1.15', '1.16' ]
+    steps:
+      - name: Checkout rest
+        uses: actions/checkout@v2
+
+      - name: Setup Go environment
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go }}
+
+      - name: Set GOPATH
+        run: echo "GOPATH=$HOME" >> $GITHUB_ENV
+      - name: Set GOBIN
+        run: echo "GOBIN=$GOPATH/bin" >> $GITHUB_ENV
+      - name: Set GO111MODULE
+        run: echo "GO111MODULE=off" >> $GITHUB_ENV
+
+      - name: Build rest
+        run: make install
+
+      - name: Run Tests
+        run: make test
+      - run: bash <(curl -s https://codecov.io/bash)
```

Comment:
I think we should remove Codecov since it [doesn't generate a report](https://github.com/sendgrid/rest/runs/4513575499?check_suite_focus=true#step:9:41) and our team doesn't use it to analyze code coverage.